### DISCUSSION TITLE
Multiple IO Processing, FX Surround Plugins

### DISCUSSION THREAD

**Discussion by jan-m01** [apohl79/audiogridder#724] [Help]
Hello Andreas,

first of all great respect for what you are doing here! 

Regarding the new multiple in and output feature: 
I'm working in surround specifically 4.0 quadro (meaning 4 outputs). I noticed that audiogridder is not correctly switching to the surround version of a FX plugins although audio gridder is getting 4 inputs. If I'm using e.g. a reverb only the stereo version of the plugin is used (displayed by the UI) and only the first 2 inputs get processed. The other left ones are getting delivered to the output unprocessed. So basically in and output is working correctly but the processing in between is not. Is this a known issue? Is there a possibility to implement this?

I hope I have explained it clearly enough. 

Best regards Jan

**Comment by apohl79** [comment]
I think I never stumbled upon this. AG it trying multiple channel configs when loading a plugin and uses the first working one. Maybe starting with more complex layouts first might be the solution. Could also be that AG is not trying all possible configs. So I guess this is doable.

Could you share the server log file? (https://audiogridder.com/bug-report-diagnostics-locations/)

Also, do you know a free plugin I could use to test? Or maybe just name a few, I might have a license for one of it.

**Reply by bsoundnz** [reply]
Hello.  Just to report I'm also finding this on my system with AG server and Windows surround VST3 plugins - potential plugin higher I/O is not detected.  I think starting with the most complex layouts instead of accepting the first working one may be the better way if it's possible.  The dev of Plogue Bidule recently tackled this with VST3.  He has said: the speaker arrangement/variable input/output switch is a bit more convoluted in VST3 than it was in VST2, requiring a bunch of code to be added for the scanning.

On the AG plugin side, I can see that it is identifying the multi I/O nicely - but that it also going using the default Juce channel names - which are just a bunch of assumptions based on the channel count, and often wrong for modern surround formats.  It may be just better to label them Ch 1, Ch 2... etc if possible, to avoid confusion.

Unfortunately, proper surround plugins are rarely free - especially in VST3.  Here's an idea, you could ask Inspired Acoustics for an NFR Inspirata Professional license.  Their amazing convolution reverb is Heavy Lifting for a PC, especially in the thick of a production.  If AG was able to support their plugin in surround they would potentially sell more licences!  The AG Scan log identifies Inspirata Pro as 2 I/O, though it supports up to 7.1.2.

Zplane PPMulator XL is another well behaved VST3 surround plugin, also utilising Juce.

Thanks for your amazing work!!!  Happy to help more in surround land...

**Comment by IzMaaR** [comment]
Same issue of multiple instances here on Samplitude X6
I hope there will be support for it one day since the summing and oversampling algos are really great in this daw compared to the others
https://www.youtube.com/watch?v=mEC_ptppwZM&ab_channel=MarcoPrimeau
this is why I work with it for a decade over having tried and compared sounds of cubase and abletone that sounded rubbish at the time being that I did the comparison even though they might have improved now but I doubt they're quite at the same level. Only sphynx pyramix was.
https://community.samplitude.com/topic/30185-dsp-versus-actual-cpu-usage/page/2/?tab=comments#comment-316026

I've just reached Samplitude support to see how it could work with audiogridder in the future. and linked this thread

**Reply by bsoundnz** [reply]
Sequoia 16 here (same engine as Samp X6).  Can also confirm I was getting multiple instances.  However I'm only attempting to run one plugin (the supremely weighty Inspirata Pro at 96K in surround), so I changed the AG server cfg file to only show plugin UI locally, as switching over to tweak on RDP is no big deal for me.  But would be happy to help future test and sort out this multiple instance issue on Samplitude/Sequoia.

**Comment by DMDComposer** [comment]
Has this been resolved? I'm sending 5.1 out to AGridder using a Fabfilter PRO-L2, however L2 only see's it as stereo. Is the only solution to use a multi-mono in AGridder?

Cheers,
DMDComposer
